src/components/functional_module/include/functional_module/generic_module.h

```diff
@@ -42,6 +42,16 @@
 
 namespace functional_modules {
 
+/**
+ * @brief The SDLEvent enum defines SDL event that module can react on
+ */
+enum SDLEvent {
```

Comment:
@dev-gh looks like all vales related to appllicaiton. And require transfer applicaiton.
I propose to rename it to ApplicaitonEvent. Because some other events that plugin may need could require other arguments

Reply:
New events can require other parameters. This one is Application events, so it has onle one parameter - application. SDL event can has a lot of different aprameters, so it should use another callbacks and events naming

Reply:
@LuxoftAKutsan 
I suggest not to consider all possible cases which can happen in future. Maybe happen.